### PR TITLE
VideoCLIP evaluation script

### PR DESCRIPTION
**Pull request by sophiazhi**
Summary:
Add script for videoCLIP evaluation from checkpoint. Switch from `argparse` to `omegaconf` in [PR #233](https://github.com/facebookresearch/multimodal/pull/233).

Test plan:
`python eval.py --use_manual_annotation --get_text_desc --get_game_frame --fixed_start_idx`
``` 
/data/home/szhi/torchmultimodal/examples/mugen/retrieval/video_clip.py:201: UserWarning: `text_trainable` acts as True when `text_pretrained` is False.
  warnings.warn("`text_trainable` acts as True when `text_pretrained` is False.")
/data/home/szhi/torchmultimodal/examples/mugen/retrieval/video_clip.py:217: UserWarning: `video_trainable` acts as True when `video_pretrained` is False.
  warnings.warn(
Loading pretrained DistilBERT from distilbert-base-uncased.
Some weights of the model checkpoint at distilbert-base-uncased were not used when initializing DistilBertModel: ['vocab_projector.bias', 'vocab_layer_norm.bias', 'vocab_projector.weight', 'vocab_transform.bias', 'vocab_transform.weight', 'vocab_layer_norm.weight']
- This IS expected if you are initializing DistilBertModel from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing DistilBertModel from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
Loading pretrained video encoder from https://pytorch.s3.amazonaws.com/models/multimodal/mugen/S3D_kinetics400.pt.
/data/home/szhi/miniconda3/envs/torch-multimodal/lib/python3.8/site-packages/pytorch_lightning/loops/utilities.py:91: PossibleUserWarning: `max_epochs` was not set. Setting it to 1000 epochs. To train without an epoch limit, set `max_epochs=-1`.
  rank_zero_warn(
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
LOADING FROM JSON FROM datasets/coinrun/coinrun_dataset_jsons/release/test.json...
NUMBER OF FILES LOADED: 100
Testing DataLoader 0: 100%|█████████████████████████████████████████████| 7/7 [00:36<00:00,  5.21s/it]
──────────────────────────────────────────────────────────────────────────────────────────────────────
              Test metric                           DataLoader 0
──────────────────────────────────────────────────────────────────────────────────────────────────────
Recall@1 (text query, video retrieval)           0.6299999952316284
Recall@1 (video query, text retrieval)           0.6899999976158142
Recall@10 (text query, video retrieval)          0.9399999976158142
Recall@10 (video query, text retrieval)          0.9399999976158142
Recall@5 (text query, video retrieval)           0.8899999856948853
Recall@5 (video query, text retrieval)           0.9300000071525574
──────────────────────────────────────────────────────────────────────────────────────────────────────
```

### CODE REVIEW

**Review by langong347 — COMMENTED** [review]
Change `evaluation.py` to `eval.py`.

LGTM but we should consider moving to the yaml file as the immediate next step.
Would request review from others since they may have more experience with writing evaluation/training loops.

File: examples/mugen/retrieval/eval.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+def parse_all_args():
+    parser = argparse.ArgumentParser()
+    for arguments_dict in arg_structure.values():
+        for arg_name, arg_options in arguments_dict.items():
+            parser.add_argument("--" + arg_name, **arg_options)
+    args = parser.parse_args()
+    return args
+
+
+def get_args_from_group(args, group_name):
```
Comment (ebsmothers) [inline examples/mugen/retrieval/eval.py:82]: Agree with @langong347 on moving to YAML if possible, this method is probably unnecessary. You can also take a look at `vqa.yaml` and `finetune.py` in #224 as an example of how to make this change.

File: examples/mugen/retrieval/eval.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        **vars(datamodule_args),
+    )
+
+    model = VideoCLIPLightningModule(
+        **vars(lightningmodule_args), **vars(videoclip_args)
+    )
+    model = model.load_from_checkpoint(evaluation_args.checkpoint_path)
+
+    trainer = Trainer(
+        accelerator=evaluation_args.accelerator, devices=evaluation_args.devices
```
Comment (ebsmothers) [inline examples/mugen/retrieval/eval.py:114]: Are these likely to be modified? (Put another way, have you tested the eval script on multiple devices?) If not you may not want to make this configurable

**Review by ebsmothers — COMMENTED** [review]
Looks good overall, just a few comments

File: examples/mugen/retrieval/eval.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+def evaluate():
+    args = parse_all_args()
+    datamodule_args = get_args_from_group(args, "datamodule")
+    lightningmodule_args = get_args_from_group(args, "lightningmodule")
+    dataset_args = get_args_from_group(args, "dataset")
+    videoclip_args = get_args_from_group(args, "videoclip")
+    evaluation_args = get_args_from_group(args, "evaluation")
+
+    dataset_args = MUGENDatasetArgs(get_audio=False, **vars(dataset_args))
```
[the remark this answers is not in the document]
Comment (sophiazhi) [reply examples/mugen/retrieval/eval.py:100]: I wanted to remove the option to retrieve audio data because videoclip uses only video and text. the datamodule returns audio separately from video if `get_audio=True`, so the extra data fetched is useless to videoclip

File: examples/mugen/retrieval/eval.py (hunk 2 above)
Comment (sophiazhi) [reply examples/mugen/retrieval/eval.py:114]: did some testing + checked [lightning docs](https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#testing:~:text=%3Dtest_dataloader)-,NOTE,single%20device%20to%20ensure%20each%20sample/batch%20gets%20evaluated%20exactly%20once.,-This%20is%20helpful) and will change to `accelerator="auto", devices=1`. i could also change accelerator to "cpu" but i think that will be an inconvenience when using the full test set

File: examples/mugen/retrieval/eval.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (sophiazhi) [reply examples/mugen/retrieval/eval.py:98]: I might add it back for the yaml config actually
